account_invoice_show_currency_rate/models/account_move.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    currency_rate_amount = fields.Float(
+        string="Rate amount", compute="_compute_currency_rate_amount", digits=0,
+    )
+    show_currency_rate_amount = fields.Boolean(
+        compute="_compute_show_currency_rate_amount", readonly=True
+    )
+
+    @api.depends(
+        "state",
+        "date",
+        "line_ids.amount_currency",
+        "company_id",
+        "currency_id",
+        "show_currency_rate_amount",
+    )
+    def _compute_currency_rate_amount(self):
+        """ It's necessary to define value according to some cases:
+        - Case A: Currency is equal to company currency (Value = 1)
+        - Case B: Move exist previously (posted) and get real rate according to lines
+        - Case C: Get expected rate (according to date) to show some value in creation.
+        """
+        self.currency_rate_amount = 1
+        for item in self.filtered("show_currency_rate_amount"):
+            lines = item.line_ids.filtered(lambda x: x.amount_currency > 0)
+            if item.state == "posted" and lines:
+                amount_currency_positive = sum(lines.mapped("amount_currency"))
+                total_debit = sum(item.line_ids.mapped("debit"))
+                item.currency_rate_amount = amount_currency_positive / total_debit
+            elif not lines or item.state != "posted":
```

Comment:
```suggestion
            else:
```

Reply:
Thanks, but is not correct because we ONLY change "default value" (1) in these case.

Reply:
Ok, but this condition is the negation of the one in the `if`, so the result will be the same, right?

Reply:
Not exactly, in these case we only apply when not lines OR state != 'posted', is not else.

Reply:
Actually it is the same, so better just put the else.

Reply:
Ok, but your condition `not lines or item.state != "posted"` is the [logic negation](https://en.wikipedia.org/wiki/Negation) of `item.state == "posted" and lines`, so the result here will be the same. Changing it to `else` avoids the extra check in the code. If this is not the desired behavior, then the current condition might be wrong...